src/main/java/tutorease/address/logic/parser/AddContactCommandParser.java

```diff
@@ -51,7 +52,13 @@ public AddContactCommand parse(String args) throws ParseException {
         Role role = ParserUtil.parseRole(argMultimap.getValue(PREFIX_ROLE).get());
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
-        Person person = new Student(name, phone, email, address, role, tagList);
+        Person person;
+
+        if (role.equals(Role.STUDENT)) {
+            person = new Student(name, phone, email, address, role, tagList);
+        } else {
+            person = new Guardian(name, phone, email, address, role, tagList);
+        }
```

Comment:
Perhaps this can be another conditional with the else throwing an exception.

Reply:
i understand, however there is a check happening within the Role class already. it will not allow any other roles other than "Student" or "Guardian"

Reply:
We should follow this example from the textbook 👍 
![image](https://github.com/user-attachments/assets/17f3fd34-f018-4b84-b78e-71f22319538c)

Reply:
Fixed! 😄